src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -30,64 +29,51 @@ public class TableStorageScaleMetricsRepository : IScaleMetricsRepository
         private const int MetricsPurgeDelaySeconds = 30;
         private const int DefaultTableCreationRetries = 3;
 
-        private readonly IConfiguration _configuration;
         private readonly IHostIdProvider _hostIdProvider;
+        private readonly IAzureTableStorageProvider _azureTableStorageProvider;
         private readonly ScaleOptions _scaleOptions;
         private readonly ILogger _logger;
         private readonly int _tableCreationRetries;
-        private readonly IDelegatingHandlerProvider _delegatingHandlerProvider;
-        private CloudTableClient _tableClient;
+        private TableServiceClient _tableServiceClient;
 
-        public TableStorageScaleMetricsRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IOptions<ScaleOptions> scaleOptions, ILoggerFactory loggerFactory, IEnvironment environment)
-            : this(configuration, hostIdProvider, scaleOptions, loggerFactory, DefaultTableCreationRetries, new DefaultDelegatingHandlerProvider(environment))
+        public TableStorageScaleMetricsRepository(IHostIdProvider hostIdProvider, IOptions<ScaleOptions> scaleOptions, ILoggerFactory loggerFactory, IAzureTableStorageProvider azureTableStorageProvider)
+            : this(hostIdProvider, scaleOptions, loggerFactory, azureTableStorageProvider, DefaultTableCreationRetries)
         {
         }
 
-        internal TableStorageScaleMetricsRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IOptions<ScaleOptions> scaleOptions, ILoggerFactory loggerFactory,
-            int tableCreationRetries, IDelegatingHandlerProvider delegatingHandlerProvider)
+        internal TableStorageScaleMetricsRepository(IHostIdProvider hostIdProvider, IOptions<ScaleOptions> scaleOptions, ILoggerFactory loggerFactory,
+            IAzureTableStorageProvider azureTableStorageProvider, int tableCreationRetries)
         {
-            _configuration = configuration;
             _hostIdProvider = hostIdProvider;
+            _azureTableStorageProvider = azureTableStorageProvider;
             _scaleOptions = scaleOptions.Value;
             _logger = loggerFactory.CreateLogger<TableStorageScaleMetricsRepository>();
             _tableCreationRetries = tableCreationRetries;
-            _delegatingHandlerProvider = delegatingHandlerProvider ?? throw new ArgumentNullException(nameof(delegatingHandlerProvider));
         }
 
-        internal CloudTableClient TableClient
+        internal TableServiceClient TableServiceClient
         {
             get
             {
-                if (_tableClient == null)
+                if (_tableServiceClient is null && !_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableServiceClient))
```

Comment:
It would be really good here to do a _try create and log failure just once_, to avoid logging the error every time the property is accessed. 

`TableStorageScaleMetricsRepository` is registered as singleton, `ScriptHost `level if I'm not mistaken: https://github.com/Azure/azure-functions-host/blob/17bad4fd6ac4a00a0d703d8dbcdbf49b221dcfe5/src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs#L149

Does that mean that specialization wouldn't affect here and we're safe assuming that the `IConfiguration` used by `IAzureTableStorageProvider` has the right entries any time this property is accessed? Just trying to avoid having a repository instance not initialized due to checking and marking as invalid too soon.

cc: @fabiocav